pkg/rtc/room.go

```diff
@@ -690,28 +690,7 @@ func (r *Room) SyncState(participant types.LocalParticipant, state *livekit.Sync
 	// can be set before addding subscriptions,
 	// in fact it is done before so that setting can be updated immediately upon subscription.
 	for _, trackSid := range state.TrackSidsDisabled {
-		// skip if it is not present in the subscribe list
-		found := false
-		for _, sid := range state.Subscription.TrackSids {
-			if sid == trackSid {
-				found = true
-				break
-			}
-		}
-		if !found {
-		participant_tracks:
-			for _, pt := range state.Subscription.ParticipantTracks {
-				for _, sid := range pt.TrackSids {
-					if sid == trackSid {
-						found = true
-						break participant_tracks
-					}
-				}
-			}
-		}
-		if found {
-			participant.UpdateSubscribedTrackSettings(livekit.TrackID(trackSid), &livekit.UpdateTrackSettings{Disabled: true})
-		}
```

Comment:
Turns out checking this is incorrect. Depending on auto subscribe or not, `SyncState` uses the `Subscriptions` field to send
1. unsubscribed tracks -> if auto subscribe is ON
2. subscribed tracks -> if auto subscribe is OFF
Just simplifying here and setting track settings based on what client sends. Can add checks later if necessary.